### PR TITLE
[1.1.x] Configure stepper drivers per axis

### PR DESCRIPTION
**Pull request by thinkyhead**
Based on #11264 (and #11133)

* Add axis specific driver definitions to determine proper timing parameters like step pulse width, delay after asserting dir level and maximum stepper frequency.
* The user can use settings in `Configuration_adv.h` to override the parsed driver IC defaults.
* The new driver definitions are used to determine which TMC or L6470 are in use and need initialization.
* By default the Allegro A4988 drivers are enabled for X, Y, Z and E0. More on this below.
* Add new TMC macros that work by checking for a certain feature, rather than the dev checking for a list of driver types that support the feature.
* References to TRAMS removed and will simply be handled by defining the four axis as `TMC5130`. However this PR doesn't yet move Marlin to the library that supports the driver.
* Differentiating between TMC2130 in SPI configured mode and standalone mode is done by having two driver definitions; `TMC2130` and `TMC2130_STANDALONE`. The way I see it, the default use case for that driver is in configured mode and using standalone mode is a special case. In standalone mode only the stepper timings are affected and they pretty much act the same as `TMC2100` as far as Marlin is concerned.
* Updates Travis to use the new driver definition method so that they actually are tested.

---

I decided to use the Allegro drivers as the default to match what is the most common configuration in the user base. Specifying any drivers is not forced but will default to DRV8825 configuration, except for DIR delay. This could be improved, or force defining at least one driver (unless overriden in _adv.h).
I think the main question is if we know the parameters different drivers require, which should be used as the default? I think we should not use 0 delays as there is no driver compatible with such timings.
However, this doesn't take into account the natural delay there is no matter what we do.
This also leads to a more general configuration change that affects the stepper ISR. Perhaps we should no longer check if `MINIMUM_STEPPER_PULSE` is defined, because from hereon it _always_ is, and rather check if the defined pulse width is greater than 20 for example. @ejtagle ?

---

The compile time driver IC macros:

`AXIS_DRIVER_TYPE(A, T)`
Check if an axis `A` has a driver type `T` associated with it.

`HAS_DRIVER(T)`
Check if the user has driver type `T` in any of the axis enabled.

`HAS_TRINAMIC`
Check if any supported Trinamic drivers are enabled in any axis.

`AXIS_IS_TMC(A)`
Check if the specified axis `A` has a supported Trinamic driver enabled.

### CODE REVIEW

File: Marlin/Configuration_adv.h [hunk 1]
```diff
@@ -813,15 +813,17 @@
...
+ *   200 : Minimum for A4988 drivers
+ *   500 : Minimum for LV8729 drivers (guess, no info in datasheet)
+ *   650 : Minimum for DRV8825 drivers
+ *  1500 : Minimum for TB6600 drivers (guess, no info in datasheet)
+ * 15000 : Minimum for TB6560 drivers (guess, no info in datasheet)
+ *
+ * Override the default value based on the driver type set in Configuration.h.
  */
-#define MINIMUM_STEPPER_DIR_DELAY 0
+//#define MINIMUM_STEPPER_DIR_DELAY 650
```
Comment (skarcha) [inline Marlin/Configuration_adv.h:826]: If you are assuming A4988 drivers as default, `MINIMUM_STEPPER_DIR_DELAY` should be 200 as default too, doesn't it? And the same for the rest of related parameters.

Comment (thinkyhead) [reply Marlin/Configuration_adv.h]: You would think. Missed this from the original submission. I'll patch it soon.

Comment (thinkyhead) [reply Marlin/Configuration_adv.h:826]: I think it's alright. This is consistent with the previous arrangement, with A4988 (or no tuning) as the default, but DRV8825 (a very common driver) as the example for overrides.

Comment (skarcha) [reply Marlin/Configuration_adv.h:826]: Yes, you're right, sorry!

**Pull request merged**